modules/cli/src/main/scala/scala/cli/commands/Repl.scala

```diff
@@ -82,7 +82,7 @@ object Repl extends ScalaCommand[ReplOptions] {
     ): Unit = {
       val res = runRepl(
         buildOptions,
-        programArgs,
+        programArgs ++ options.shared.scalac.scalacOption,
```

Comment:
- Better use `buildOptions.scalaOptions.scalacOptions.toSeq.map(_.value.value)` rather than `options.shared.scalac.scalacOption`. The latter corresponds to CLI options, while the former also takes into account using directives.
- I don't think this is going to work for Ammonite. Better add those options in `runRepl` itself, when when we specifically run the default REPL.